fhir-database-utils/src/main/java/com/ibm/fhir/database/utils/postgres/PostgresVacuumSettingDAO.java

```diff
@@ -0,0 +1,121 @@
+        if (translator.getType() == DbType.POSTGRESQL) {
+            boolean processSettings1a = true;
+            boolean processSettings1b = true;
+            boolean processSettings2 = true;
+            LOG.fine(() -> "Checking the table vacuum settings");
+            final String statement0 = "select (reloptions)::VARCHAR "
+                    + "from pg_class "
+                    + "join pg_namespace on pg_namespace.oid = pg_class.relnamespace "
+                    + "where relname = LOWER('" + tableName + "') "
+                    + "and pg_namespace.nspname = LOWER('" + schema + "')";
+            try (PreparedStatement ps = c.prepareStatement(statement0)) {
+                ps.execute();
+                ResultSet rs = ps.getResultSet();
+                if (rs.next()) {
+                    String settings = rs.getString(1);
+                    if (!rs.wasNull()) {
+                        String[] sar = settings
+                                    .replace("{", "")
+                                    .replace("}", "")
+                                    .split(",");
+                        for (String setting : sar) {
+                            String[] kv = setting.split("=");
+                            if ("autovacuum_vacuum_scale_factor".equals(kv[0]) && vacuumScaleFactor == Double.parseDouble(kv[1])) {
+                                processSettings1a = false;
+                            } else if ("autovacuum_vacuum_threshold".equals(kv[0]) && vacuumThreshold == Integer.parseInt(kv[1])) {
+                                processSettings1b = false;
+                            } else if ("autovacuum_vacuum_cost_limit".equals(kv[0]) && vacuumCostLimit == Integer.parseInt(kv[1])) {
+                                processSettings2 = false;
+                            }
+                        }
+                    }
+                }
+            } catch (SQLException x) {
+                throw translator.translate(x);
+            }
+
+            if (processSettings1a || processSettings1b) {
+                // -- Lower the trigger threshold for starting work
+                // alter table fhirdata.logical_resources SET (autovacuum_vacuum_scale_factor = 0.01,
+                // autovacuum_vacuum_threshold=1000);
+                final String statement1 = "alter table " + schema + "." + tableName
+                        + " SET (autovacuum_vacuum_scale_factor = " + vacuumScaleFactor + ", autovacuum_vacuum_threshold=" + vacuumThreshold + ")";
+
+                LOG.fine(() -> "Updating the table " + schema + "." + tableName + " [" + vacuumScaleFactor + "," + vacuumThreshold + "]");
+                try (PreparedStatement ps = c.prepareStatement(statement1)) {
+                    ps.execute();
+                } catch (SQLException x) {
+                    throw translator.translate(x);
+                }
+            }
+
+            if (processSettings2) {
```

Comment:
If you're modifying the same table as in the block above starting at line 89, seems like it would be more efficient to just do one update for all 3 properties

Reply:
Right, I had interpreted Robin's recommendation that they should be done independently.  Let me double check the docs, and make the change if it is appropriate

Reply:
Changed